dbms/src/Flash/Pipeline/Pipeline.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Flash/Executor/ResultHandler.h>
+#include <Operators/OperatorPipeline.h>
+
+#include <deque>
+#include <vector>
+
+namespace tipb
+{
+class DAGRequest;
+}
+
+namespace DB
+{
+class Pipeline;
+using PipelinePtr = std::shared_ptr<Pipeline>;
+using Pipelines = std::vector<PipelinePtr>;
+
+class Event;
+using EventPtr = std::shared_ptr<Event>;
+using Events = std::vector<EventPtr>;
+
+class PhysicalPlanNode;
+using PhysicalPlanNodePtr = std::shared_ptr<PhysicalPlanNode>;
+
+class PipelineExecStatus;
+
+class Pipeline : public std::enable_shared_from_this<Pipeline>
+{
+public:
+    void addPlan(const PhysicalPlanNodePtr & plan);
```

Comment:
This name is somehow misleading. We should be strict on when to use `Plan` and when to use `PlanNode`. Maybe we better call it `addPlanNode`?

Reply:
ok, updated.